### ISSUE TITLE
Feature Request: Add JID (+Timestamp) into filename for received Media

### ISSUE THREAD

**Issue by aslmx** [siacs/Conversations#3589]
# Feature Request

After the suggestions to store received media in a per-contact subfolder structure have been rejected for plausible reasons

see e.g.:

https://github.com/siacs/Conversations/issues/3582
https://github.com/siacs/Conversations/issues/1540

I'd like to ask (again, but formally now) if it is possible to store the JID of the contact and optionally the timestamp in the filename, to make it easier to filter files based on those criteria.

You might argue that the timestamp is already part of a files metadata in most file systems, but depending on how you handle your files (copy to other systems) the timestamps might get messed up.

## Now:
Filename stored consists of random hex characters not possible to distinguish from which contact $media was received when just looking into the Conversations Media folders

## After Feature Request has been implemented:

Filename are named with JID of contact, Timestamp and random characters, e.g.

contact@jabberserver.tld_2019-11-22_121332_(random_hex_characters).jpeg

## Benefit

Files can be distinguished and copied/moved/deleted/backuped/etc. on a per contact base very easily.

# Extension:

- Make this configurable
- Make the pattern configurable, either by letting users specify a mask or by giving some options
(e.g. JID_Timestamp_random.extension / TImestamp_JID_random.extension ...)

**Comment by licaon-kter** [comment]
And when you share this file.....you leak the JID of the sender? :(

**Comment by aslmx** [comment]
@licaon-kter fair point. If this is not desired Conversations could randomize the outgoing filename again...

Actually I think this is already implemented. When I send "blabla_report_2019.PDF" I know it will become hexrandomvalue.PDF at the other conversations recipient... regardless from where I originally obtained the file.

This is actually quite annoying as well sometimes, but a different discussion...

(Btw: actually my desire is to distinguish files via the gallery/file manager app, I don't even need the real jid to be in the filename, just a unique but constant value as a pseudonym. But I don't think only my usecase should be covered)

**Comment by zak333** [comment]
As I am archiving most received images and renaming them manually for now I would gladly be satisfied if the file could at least be named with the timestamp of the received date+time, like 20191215_212423.jpg for example. A randomized string shouldn't be needed in most cases, and for the rare case that someone receives more than one file in the same second a suffix could be added (e.g. 20191215_212423_1.jpg).

**Comment by licaon-kter** [comment]
@zak333 Depends on how you get the files off the device, a simple **"[jhead](https://www.sentex.ca/~mwandel/jhead/) -ft \*"** takes care of the JPEGs (update internal EXIF data DateTaken with date-of-file aka when you've received it)

Why it depends on transfer? Well you want a way to get the files with their original date, eg. Syncthing works great ;)

**Comment by zak333** [comment]
@licaon-kter This issue is about not having to do manual work like this and have *some* readable and manageable filename. Additionally the timestamp makes sure that files are ordered chronologically in a common file or media browser.

And not every file includes EXIF data.

**Comment by ToEatFreshBabies** [comment]
> And when you share this file.....you leak the JID of the sender? :(

How about a locally generated (unique) hash of JID that is noise when shared

**Comment by aslmx** [comment]
> How about a locally generated (unique) hash of JID that is noise when shared

Well, that should make it pretty unique, but still non-intuitive... I wouldn't favor this solution tbh.

**Comment by DMSalesman** [comment]
Seems the original filename of files downloaded in an OMEMO-encrypted chat is visible to Conversations, as seen in the `aesgcm://` schema.

Could it be possible to save the file under that very filename, instead of mangling it?

**Comment by aslmx** [comment]
So... i'm still spending minutes every few days to sort out received media manually - rather than seconds if the filenames had a constistent pattern.

Any news or anyone motivated? thx :)

**Comment by DMSalesman** [comment]
@aslmx I run a customized build of Conversations which maintains the original filename of received media (i.e. without alterations of any kind - if the sender's file is called "evil_plan.pdf", that'll be the resulting filename on Android).

Thing is, I doubt Conversations would be interested in a PR with the change, judging as how the decision to generate a "random" name seems deliberate.
